### DISCUSSION TITLE
How to enable FastCGI in Laragon

### DISCUSSION THREAD

**Discussion by TheOldMan2000** [leokhoa/laragon#498] [Q&A] [answered]
Hi folks,

Today I updated PHPMyAdmin and PHP 8.1 on my local Win11 machine. Using the latest Laragon 6 and Apache httpd-2.4.54-win64-VS16.

Does anyone have a tutorial link or can remind me how to enable FastCGI? I seem to recall it was a simple nifty method using Laragon thanks to Leo. 

I don't really want to run it as a service if possible.
Everything works, except PHP Info is showing I'm still running in the original CGI. mode. 

I have the 2 proxy_module modules/mod_proxy.so and fproxy_fcgi_module modules/mod_proxy_fcgi.so Apache modules uncommented in httpd.conf and I found I have an outdated fcgid.conf file in my laragon/etc/apache2 folder. 

So I updated the paths to the latest PHP v8.1.19 and it now looks like this:

````LoadModule fcgid_module "C:/laragon/etc/apache2/modules/mod_fcgid-2.3.10-win64-VS16.so"
C:\laragon\bin\php\php-8.1.19-Win32-vs16-x64
<IfModule fcgid_module>
FcgidInitialEnv PATH "C:/laragon/bin/php/php-8.1.19-Win32-vs16-x64;C:/Windows/system32;C:/Windows;C:/Windows/System32/Wbem;"
FcgidInitialEnv SystemRoot "C:/Windows"
FcgidInitialEnv SystemDrive "C:"
FcgidInitialEnv TEMP "C:/Windows/Temp"
FcgidInitialEnv TMP "C:/Windows/Temp"
FcgidInitialEnv windir "C:/Windows"

# 10 hrs: in case you have long running scripts, increase FcgidIOTimeout 
FcgidIOTimeout 36000
FcgidConnectTimeout 16
FcgidMaxRequestsPerProcess 0
FcgidMaxProcesses 50
FcgidMaxRequestLen 81310720
# Location php.ini:
FcgidInitialEnv PHPRC "C:/laragon/bin/php/php-8.1.19-Win32-vs16-x64"
FcgidInitialEnv PHP_FCGI_MAX_REQUESTS 0

<Files ~ "\.php$>"
AddHandler fcgid-script .php
Options +ExecCGI
FcgidWrapper "C:/laragon/bin/php/php-8.1.19-Win32-vs16-x64/php-cgi.exe" .php
</Files>
</IfModule>````

However, although everything works, PHP Info is showing I'm running in the original CGI. mode.

**Comment by Pen-y-Fan** [comment, marked as answer]
Install the NTS version of PHP and Laragon will automatically configure Apache to use FastCGI, see my blog [How to update the PHP version in Laragon](https://pen-y-fan.github.io/2023/01/15/how-to-update-the-php-version-in-laragon/) for more details.

**Reply by t-e-s-tweb** [reply]
Thanks. Its because of people like you that learning is fun again. Thanks for the blog post and the full explanation. 👍

**Comment by TheOldMan2000** [comment]
That is awesome, thank you! Will enjoy reading the blog also.

**Comment by TheOldMan2000** [comment]
Hi @Pen-y-Fan  Thanks for the tip, I got it working. I even remembered to update the PHP paths in my chronical.dat file's list of cronjobs!
However, in my conf folder, I still have a mod_php file. Do I need to keep that?

![image](https://github.com/leokhoa/laragon/assets/13329789/d4c54ea2-f48e-4b72-ba7e-a3256382b6e5)

Inside it still points to the TS version of PHP.

```
`# This file is auto-generated, so please keep it intact.
LoadModule php_module "C:/laragon/bin/php/php-8.1.19-Win32-vs16-x64/php8apache2_4.dll"
PHPIniDir "C:/laragon/bin/php/php-8.1.19-Win32-vs16-x64"
<IfModule mime_module>
    AddType application/x-httpd-php .php
</IfModule>`
```

Many thanks.

**Comment by rivman** [comment]
Thank you for making us gain some time on this on how to activate mod_fcgid for php 8.2.21 on laragon 6.00
